### PR TITLE
Add "TextParagraphParser" and "multiple empty line parser"

### PR DESCRIPTION
**Pull request by andysim3d**
Add: 
- Text paragraph parser
- Multiple empty line parser
- Merger of quote

Amend: 
- quote parser enable "multiple line", and return correct begin & end instead of (0, len(content))
- parser logic, now it will try to remove empty lines, and use text paragraph parser to parse unmatched part. 
- regexp of horizontal rule.

### CODE REVIEW

File: src/blocks/element.py [hunk 1]
```diff
@@ -33,3 +37,6 @@ def nested(self):
         """Could this element hold other elements inside, True for yes.
         """
         return True
+
+    def __eq__(self, value):
+        return isinstance(value, Element) and value.content() == self.content()
```
Comment (jinglongwu) [inline src/blocks/element.py]: equality of different type but same content().

File: src/parser/parser.py [hunk 2]
```diff
@@ -59,15 +67,50 @@ def _invoke_parsers(self, parent, content):
...
             if begin != -1 and begin < start_index:
                 start_index, end_index, final_element = begin, end, element
-        if start_index > 0 and start_index != len(content):
-            text_paragraph = TextParagraph(content[:start_index])
-            link_parent_and_child(parent, text_paragraph)
+        if start_index > 0:
+            # try to eliminate empty lines
+            begin, end, _ = parse_empty_newlines(content)
+            if begin == 0:
+                print("here")
```
Comment (jinglongwu) [inline src/parser/parser.py]: here!

File: src/parser/parser.py [hunk 3]
```diff
@@ -59,15 +67,50 @@ def _invoke_parsers(self, parent, content):
...
+                ## put an empty text paragraph to avoid paragraph merge.
+                start_index, end_index, final_element = begin, end, TextParagraph(
+                    "")
+                ## put an empty text paragraph to avoid paragraph merge.
+            else:
+                print("here2")
+                start_index, end_index, final_element = self._default(
+                    content[:start_index])
+        print("parsed object", content, final_element)
```
Comment (jinglongwu) [inline src/parser/parser.py]: should we consider some debug log?

File: src/blocks/element.py (hunk 1 above)
Comment (andysim3d) [reply src/blocks/element.py]: done

File: src/parser/parser.py (hunk 2 above)
Comment (andysim3d) [reply src/parser/parser.py]: done

File: src/parser/parser.py (hunk 3 above)
Comment (andysim3d) [reply src/parser/parser.py]: we can.... but I'll remove them first

**Review by jinglongwu — APPROVED** [review]
LGTM

**Pull request merged**